src/operations/update.ts

```diff
@@ -29,12 +29,12 @@ export interface UpdateResult {
   acknowledged: boolean;
   /** The number of documents that matched the filter */
   matchedCount: number;
-  /** The number of documents that were modified */
+  /** The number of documents ∂that were modified */
   modifiedCount: number;
   /** The number of documents that were upserted */
   upsertedCount: number;
   /** The identifier of the inserted document if an upsert took place */
-  upsertedId: ObjectId;
+  upsertedId: ObjectId | null;
```

Comment:
This doesn't account for [setOnInsert](https://www.mongodb.com/docs/v6.0/reference/operator/update/setOnInsert/#example)

Reply:
What are you saying the correct type should be?  Either way, we have no way of knowing what the actual id type is.  Should we type it as unknown?

Reply:
Both the filter for the operation and the `$set` and `$setOnInsert` operators match against `TSchema`, meaning that if an `_id` is passed it at all, it would be constrained to `InferIdType`, so it would make sense to do that here, too. It's true that this does not account for the `_id` being missing entirely resulting in the `ObjectId` being generated server-side, but this is a similar issue to what we have in, for example, `insertOne` and `insertMany` (if the user does not provide an `_id` or a `pkFactory`), with the only difference being that a `pkFactory` can't be applied here as a backup. So if we follow our existing pattern, we would call the use of the upsert without specifying an `_id` in the case of a custom `_id` `TSchema` "incorrect usage", and advise the users in our documentation because TypeScript can't effectively catch that. Our operating assumption is that users performing upserts do not want to introduce inconsistencies into their schema, if one exists. But we can discuss that further with the team if you believe there is a stronger argument to be made for `unknown`.